Objects/typeobject.c

```diff
@@ -461,7 +461,7 @@ static PyMemberDef type_members[] = {
 static int
 check_set_special_type_attr(PyTypeObject *type, PyObject *value, const char *name)
 {
-    if (!(type->tp_flags & Py_TPFLAGS_HEAPTYPE)) {
+    if (_PyType_HasFeature(type, Py_TPFLAGS_IMMUTABLETYPE)) {
         PyErr_Format(PyExc_TypeError,
                      "can't set %s.%s", type->tp_name, name);
```

Comment:
I suggest to advertize the new Py_TPFLAGS_IMMUTABLETYPE flag in the error message ;-)

```suggestion
                     "can't set '%r' attribute of immutable type %s", name, type->tp_name);
```

Same remark for type_set_annotations(). Please also change type_setattro() error message.

Reply:
Will do!

Reply:
Improved comments in fd929c014bf0620748c25c32378505db69fde58f :)